src/Microsoft.FeatureManagement/ConfigurationFeatureDefinitionProvider.cs

```diff
@@ -193,64 +251,197 @@ We support
+                    {
+                        Name = "AlwaysOn"
+                    });
+                }
+            }
+
+            return new FeatureDefinition()
+            {
+                Name = featureName,
+                EnabledFor = enabledFor,
+                RequirementType = requirementType
+            };
+        }
+
+        private string GetFeatureFlagSectionName(IConfigurationSection section)
+        {
+            if (_azureAppConfigurationFeatureFlagSchemaEnabled)
             {
-                return _configuration.GetChildren();
+                return section[AzureAppConfigurationFeatureFlagFields.Id];
             }
 
-            Logger?.LogDebug($"No configuration section named '{FeatureManagementSectionName}' was found.");
+            return section.Key;
+        }
+
+        private IEnumerable<IConfigurationSection> GetFeatureDefinitionSections()
+        {
+            if (!_configuration.GetChildren().Any())
+            {
+                return Enumerable.Empty<IConfigurationSection>();
+            }
+
+            bool hasFeatureManagementSection = _configuration.GetChildren()
+                .Any(section => string.Equals(section.Key, ConfigurationFields.FeatureManagementSectionName, StringComparison.OrdinalIgnoreCase));
+
+            if (!hasFeatureManagementSection && !RootConfigurationFallbackEnabled)
+            {
+                return Enumerable.Empty<IConfigurationSection>();
+            }
+
+            IConfiguration featureManagementConfigurationSection = hasFeatureManagementSection ?
+                _configuration.GetSection(ConfigurationFields.FeatureManagementSectionName) :
+                _configuration;
+
+            if (_azureAppConfigurationFeatureFlagSchemaEnabled)
+            {
+                IConfigurationSection featureFlagsSection = featureManagementConfigurationSection.GetSection(AzureAppConfigurationFeatureFlagFields.FeatureFlagsSectionName);
+
+                return featureFlagsSection.GetChildren();
+            }
+
+            return featureManagementConfigurationSection.GetChildren();
+        }
+
+        private static bool HasAzureAppConfigurationFeatureFlagSchema(IConfiguration featureManagementConfiguration)
+        {
+            bool hasFeatureFlagsSection = featureManagementConfiguration.GetChildren()
+                .Any(section => string.Equals(section.Key, AzureAppConfigurationFeatureFlagFields.FeatureFlagsSectionName, StringComparison.OrdinalIgnoreCase));
+
+            if (hasFeatureFlagsSection)
+            {
+                IConfigurationSection featureFlagsConfigurationSection = featureManagementConfiguration.GetSection(AzureAppConfigurationFeatureFlagFields.FeatureFlagsSectionName);
+
+                //
+                // FeatureFlags section is an array
+                return string.IsNullOrEmpty(featureFlagsConfigurationSection.Value) && 
+                    featureFlagsConfigurationSection.GetChildren()
```

Comment:
If `featureFlagsConfigurationSection.GetChildren()` returns empty, then `All` would return true. Seems unintended.

Reply:
This is intended. If there is "FeatureFlags" section, either it is empty or an array, we consider it as using the App Config schema.